### ISSUE TITLE
can we report something about the modules in use?

### ISSUE THREAD

**Issue by sam-github** [nodejs/node-report#19]
A full `npm ls` tree would be too much, but would there be value in reporting at least any binary addons that have been loaded? I guess if addons fault they will show up in the stack traces, but they may also be sources of more subtle corruption (writes to stray pointers corrupting heap, for example).

**Comment by rnchamberlain** [comment]
Yes this would be a useful enhancement. We could do something a bit more general, and list all binaries from /proc/pid/maps, e.g.:

```
000000400000-000001bbe000 r-xp 08:01 2886147    /home/user/node-v6.9.1-linux-x64/bin/node
7fda2dfc4000-7fda2dfcf000 r-xp 08:01 3015273    /home/user/test/node_modules/nodereport/build/Release/nodereport.node
7fda301d4000-7fda30393000 r-xp 08:01 3281614    /lib/x86_64-linux-gnu/libc-2.23.so
7fda3059d000-7fda305b5000 r-xp 08:01 3281610    /lib/x86_64-linux-gnu/libpthread-2.23.so
7fda307ba000-7fda307d0000 r-xp 08:01 3281420    /lib/x86_64-linux-gnu/libgcc_s.so.1
7fda30cd9000-7fda30e4b000 r-xp 08:01 529796     /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.21
7fda3105b000-7fda31062000 r-xp 08:01 3281577    /lib/x86_64-linux-gnu/librt-2.23.so
```

**Comment by mhdawson** [comment]
+1 from me on this.

**Comment by hhellyer** [comment]
I've added PR #38 to try to address this, it asks the OS for the list of loaded modules/libraries rather than just looking at the modules node has loaded.

**Comment by rnchamberlain** [comment]
@sam-github this is now available in node-report v2.0.0, the list of loaded libraries appears at the end of the report

**Comment by sam-github** [comment]
Great
